### ISSUE TITLE
DaemonSet updates landed in 1.6

### ISSUE THREAD

**Issue by errordeveloper** [weaveworks/weave#2879]
https://github.com/kubernetes/features/issues/124 has landed in 1.6 (via https://github.com/kubernetes/kubernetes/pull/41116).

Documentation: https://kubernetes.io/docs/tasks/manage-daemon/update-daemon-set/

Right now our DaemonSet defaults to `.spec.updateStrategy.type: OnDelete` (you can see it by reading the object from a cluster with `kubectl get ds -n kube-system -o yaml weave-net`), we should test how `RollingUpdate` would work and document the process. This is orthogonal to launch generator. I'm happy to include rolling update as an optional feature for testing.

**Comment by bboreham** [comment]
@errordeveloper are you saying you want to make the change and test it, or you want someone else to code up the change and you will test it?

From my reading of the docs, this would enable us to remove the instructions about carefully deleting one pod at a time (when using 1.6)

**Comment by errordeveloper** [comment]
It would be great if someone else could test it first.

**Comment by errordeveloper** [comment]
Because this may have implications.

**Comment by mikebryant** [comment]
We're planning to turn rolling updates on for everything, including weave. Will report back after 1.6.2 is released. (Our upgrade to 1.6 is currently blocked because weave doesn't work at all, due to https://github.com/kubernetes/kubernetes/issues/44041)

**Comment by mikebryant** [comment]
We're currently running a mixed-version cluster. (We've let the kubelets lag at 1.5.4), due to the hostpid issue. (And apparently this scenario is supported)
The daemonset upgrades haven't caused us any problems with weave. We changed the manifest from 1.9.3 to 1.9.4 and watched the upgrade roll out without incident.

Note: we have been running a modified DaemonSet, to cope better with large-scale temporary node loss: 
```
      volumes:
      - hostPath:
          path: /var/lib/weave-kube
        name: weavedb
```

**Comment by bboreham** [comment]
Fixed in #3024
